go/worker/common/committee/group.go

```diff
@@ -609,13 +503,8 @@ func (g *Group) PublishScheduledBatch(
 		return nil, fmt.Errorf("group: not leader of txn scheduler committee")
 	}
 
-	xc := g.activeEpoch.executorCommittees[committeeID]
-	if xc == nil {
-		return nil, fmt.Errorf("group: invalid executor committee")
-	}
-
+	// XXX: this should be moved to the txn scheduler and passed as argument
```

Comment:
not just constructing this message, this whole method seems very specialized, e.g. with checking membership as txn scheduler leader

Reply:
Yeah, we should probably just expose a more generic `Publish`. I was planning to do that but forgot, thanks for finding this comment ;)

Reply:
Fixed.